File: packages/square-mobile-payments/ios/Plugin/Classes/Options/AuthorizeOptions.swift
import Foundation
import Capacitor

@objc public class AuthorizeOptions: NSObject {
    let accessToken: String

    init(_ call: CAPPluginCall) throws {
        self.accessToken = try AuthorizeOptions.getAccessTokenFromCall(call)
    }

    private static func getAccessTokenFromCall(_ call: CAPPluginCall) throws -> String {
        guard let accessToken = call.getString("accessToken") else {
            throw CustomError.accessTokenMissing
        }
        return accessToken
    }
}

File: packages/square-mobile-payments/android/src/main/java/io/capawesome/capacitorjs/plugins/squaremobilepayments/classes/options/AuthorizeOptions.java
package io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.CustomExceptions;

public class AuthorizeOptions {

    @NonNull
    private final String accessToken;

    public AuthorizeOptions(@NonNull PluginCall call) throws Exception {
        this.accessToken = AuthorizeOptions.getAccessTokenFromCall(call);
    }

    @NonNull
    public String getAccessToken() {
        return accessToken;
    }

    @NonNull
    private static String getAccessTokenFromCall(@NonNull PluginCall call) throws Exception {
        String accessToken = call.getString("accessToken");
        if (accessToken == null) {
            throw CustomExceptions.ACCESS_TOKEN_MISSING;
        }
        return accessToken;
    }
}
